src/v2/controllers/FormController.js

```diff
@@ -283,4 +287,30 @@ export default Controller.extend({
     button.toggleClass('link-button-disabled', disabled);
   },
 
+  /**
+    * When "Remember My Username" is enabled, we save the identifier in a cookie
+    * so that the next time the user visits the SIW, the identifier field can be 
+    * pre-filled with this value.
+   */
+  updateIdentifierCookie: function(model) {
+    const formName = model.get('formName');
+
+    // Only update the cookie when the user has successfully identified themselves (i.e. if we're on the
+    // challenge-authenticator form) to avoid incorrect/uneccessary updates.
```

Comment:
can you confirm with backend that this is always the case? that.. we present `challenge` view always only after authentication. Is there any corner case ?

Reply:
Yeah from my understanding this is the case, but I would like to confirm with @iancorcoran-okta as well. Ian the intention here is to update the cookie only after successful authentication and here I was relying on the successful response when showing the challenge-authenticator form. Is this approach correct?

Reply:
To make sure I'm understanding, you're asserting that the identifier is valid iff you get a success response from `challenge-authenticator`? I don't think that works... e.g. if an app only requires 1FA and user authenticates via IdP,  or may be some cases around user enumeration

Reply:
@iancorcoran-okta Yes essentially on the client side I need to know that successful authentication has happened, and I was relying on the above which seems incorrect now. The question is I guess how can I determine that on the client?

Reply:
This is reminding me of the `processCreds` problem, where you need the API to return the identifier on identify with password remediation success?

Reply:
Note: moved this conversation to this slack thread: https://okta.slack.com/archives/CCA77QVL2/p1629477508430300

Reply:
UPDATE: as per the slack thread above, the most reliable way is to check for the `success` or `successWithInteractionCode` objects in the response. The implementation has been updated now